src/rdkafka_offset.c

```diff
@@ -173,7 +173,7 @@ static int64_t rd_kafka_offset_file_read (rd_kafka_toppar_t *rktp) {
 	char buf[22];
 	char *end;
 	int64_t offset;
-	int r;
+	size_t r;
```

Comment:
The `if ((r = fread(...)) == -1)` check can probably be removed completely. Just do  `r = fread(..);` and let the sub-sequent `r == 0` handle it (as is)